Upgrade to repo-config v0.13.5

llucax opened the pull request.

llucax committed:
Update files to repo-config v0.12

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Update files to repo-config v0.13

We had to merge the `ci.yaml` file that normally comes with repo-config
to the `python-ci.yaml` file, and we renamed `ci-pr.yaml` to
`python-ci-pr.yaml`.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Re-format pyproject.toml using taplo

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Bump repo-config to v0.13.5

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Bump mkdocstrings deps to be compatible with repo-config

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
This seems to be failing because the `selection` key in the [mkdocstrings config](https://github.com/frequenz-floss/frequenz-resampling-rs/blob/ce84d66d9c3841512b344dec921cd5976b702f2b/mkdocs.yml#L114-L116) doesn't seem to be supported. I searched a lot looking for it or alternatives and see no mention at all in mkdocstrings docs, so I wonder how this is currently working. Do you remember anything about this @phillip-wenig-frequenz ?

phillip-wenig-frequenz commented:
> This seems to be failing because the `selection` key in the [mkdocstrings config](https://github.com/frequenz-floss/frequenz-resampling-rs/blob/ce84d66d9c3841512b344dec921cd5976b702f2b/mkdocs.yml#L114-L116) doesn't seem to be supported. I searched a lot looking for it or alternatives and see no mention at all in mkdocstrings docs, so I wonder how this is currently working. Do you remember anything about this @phillip-wenig-frequenz ?

I'll have a look

llucax committed:
Use the proper path to sources when generating docs

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Bump mkdocstrings dependencies

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Bump pyo3 dependency to 0.26

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Bump frequenz-floss/gh-action-setup-python-with-deps to v1.0.1

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Remove paths options for mkdocstrings handler

This option was wrongly located (it is not an "option", but a
configuration to the "python" handler directly).

We remove it anyways becasue the default is "." and we should actually
use ".", as the "frequenz" package is created there.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Remove wrong "selection" section

This was never for this, is an old name for "options", but the "include"
key was never available.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
I spent some time revamping this PR, I added some more dependency updates and fixes to generate docs. It is weird because I can't get the docs to build locally, but it somehow works in the CI.

<details>
<summary>Error</summary>

```
$ mkdocs build
INFO    -  FutureWarning: The 'import' key is renamed 'inventories' for the Python handler
             File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings/_internal/handlers/base.py", line 716, in get_handler
               self._handlers[name] = module.get_handler(**kwargs)
             File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings_handlers/python/_internal/handler.py", line 419, in get_handler
               warn("The 'import' key is renamed 'inventories' for the Python handler", FutureWarning, stacklevel=1)
INFO    -  [macros] - Preinstalled modules:  frequenz.repo.config.mkdocs.mkdocstrings_macros
INFO    -  [macros] - Found external Python module 'frequenz.repo.config.mkdocs.mkdocstrings_macros' in: /home/luca/devel/resampling-rs
Failed to get version info: Environment variable 'GITHUB_REPO' is not defined
INFO    -  [macros] - Functions found: define_env,on_pre_page_macros,on_post_page_macros,on_post_build
INFO    -  [macros] - No default module `main` found
INFO    -  [macros] - Config variables: ['extra', 'config', 'environment', 'plugin', 'git', 'social', 'version', 'code_annotation_marker', 'version_requirement', 'macros', 'filters', 'filters_builtin']
INFO    -  [macros] - Config macros: ['context', 'macros_info', 'now', 'fix_url']
INFO    -  [macros] - Config filters: ['pretty', 'relative_url', 'slugify']
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /home/luca/devel/resampling-rs/site
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
             - reference/SUMMARY.md
             - reference/noxfile.md
             - reference/frequenz/resampling/index.md
             - reference/tests/index.md
             - reference/tests/test_resampler.md
ERROR   -  Error reading page 'reference.md': Could not resolve alias frequenz.resampling._rust_backend.Resampler pointing at builtins.Resampler (in frequenz/resampling/_rust_backend.cpython-311-x86_64-linux-gnu.so:None)
Traceback (most recent call last):
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 2142, in _resolve_target
    resolved = self.modules_collection.get_member(self.target_path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/mixins.py", line 79, in get_member
    return self.members[parts[0]].get_member(parts[1:])  # type: ignore[attr-defined]
           ~~~~~~~~~~~~^^^^^^^^^^
KeyError: 'builtins'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/bin/mkdocs", line 10, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocs/__main__.py", line 288, in build_command
    build.build(cfg, dirty=not clean)
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 310, in build
    _populate_page(file.page, config, files, dirty)
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 167, in _populate_page
    page.render(config, files)
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocs/structure/pages.py", line 285, in render
    self.content = md.convert(self.markdown)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/markdown/core.py", line 357, in convert
    root = self.parser.parseDocument(self.lines).getroot()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 117, in parseDocument
    self.parseChunk(self.root, '\n'.join(lines))
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 136, in parseChunk
    self.parseBlocks(parent, text.split('\n\n'))
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 158, in parseBlocks
    if processor.run(parent, blocks) is not False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings/_internal/extension.py", line 128, in run
    html, handler, _ = self._process_block(identifier, block, heading_level)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings/_internal/extension.py", line 207, in _process_block
    rendered = render(data, options)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings_handlers/python/_internal/handler.py", line 292, in render
    template_name = rendering.do_get_template(self.env, data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/mkdocstrings_handlers/python/_internal/rendering.py", line 660, in do_get_template
    extra_data = getattr(obj, "extra", {}).get("mkdocstrings", {})
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 1584, in extra
    return self.final_target.extra
           ^^^^^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 2104, in final_target
    target = target.target  # type: ignore[assignment]
             ^^^^^^^^^^^^^
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 2069, in target
    self.resolve_target()
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 2136, in resolve_target
    self._resolve_target()
  File "/home/luca/devel/resampling-rs/.direnv/python-3.11/lib/python3.11/site-packages/griffe/_internal/models.py", line 2144, in _resolve_target
    raise AliasResolutionError(self) from error
griffe._internal.exceptions.AliasResolutionError: Could not resolve alias frequenz.resampling._rust_backend.Resampler pointing at builtins.Resampler (in frequenz/resampling/_rust_backend.cpython-311-x86_64-linux-gnu.so:None)
```

</details>
